lib/ssl/src/ssl_handshake.erl

```diff
@@ -2558,6 +2590,12 @@ decode_extensions(<<?UINT16(?SNI_EXT), ?UINT16(Len),
     decode_extensions(Rest, Version, MessageType,
                       Acc#{sni => dec_sni(NameList)});
 
+decode_extensions(<<?UINT16(?MFL_EXT), ?UINT16(1), ?BYTE(MFL), Rest/binary>>,
```

Comment:
> If a server receives a maximum fragment length negotiation request    for a value other than the allowed values, it MUST abort the  handshake with an "illegal_parameter" alert.

If we receive a faulty value the whole extension will be dropped but we should instead send an illegal_parameter alert.